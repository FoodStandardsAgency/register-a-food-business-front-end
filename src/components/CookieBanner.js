import styled from "react-emotion";
import { Paragraph, Button, asAnchor } from "govuk-react";

const fontSize = "16px";

const Banner = styled("div")`
  background-color: #d9e8f2;
`;

const BannerBody = styled("div")`
  width: calc(100% - 30px);
  max-width: 960px;
  margin: auto;
  padding: 15px 0;

  @media (min-width: 641px) {
    width: calc(100% - 60px);
  }
`;

const BannerParagraph = styled(Paragraph)`
  font-size: ${fontSize};
  @media only screen and (min-width: 641px) {
    font-size: ${fontSize};
  }
`;

const BannerRow = styled("div")`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: flex-start;
  @media only screen and (min-width: 641px) {
    justify-content: space-between;
    flex-direction: row;
    align-items: flex-end;
  }
`;

const BannerActionContainer = styled("div")`
  display: flex;
  align-items: flex-start;
  margin-top: 10px;
  justify-content: space-between;
  width: 100%;
  @media only screen and (min-width: 641px) {
    flex-wrap: nowrap;
    justify-content: flex-end;
    margin-top: 0px;
    align-items: flex-end;
  }
`;

const BannerLinkContainer = styled("div")`
  display: flex;
  flex-direction: column;
  flex-shrink: 0;
  margin-right: 20px;
`;

const CookieButton = styled(Button)`
  margin-bottom: 0px;
  background-color: white;
  box-shadow: 0 2px 0 #4e80b4;
  color: #0b0c0c;
  &:hover {
    background-color: #f2f2f2;
    color: #0b0c0c;
  }
  &:focus {
    background-color: #f2f2f2;
    color: #0b0c0c;
  }
  font-size: ${fontSize};
  @media only screen and (min-width: 641px) {
    font-size: ${fontSize};
    margin-bottom: 0px;
  }
`;

const RejectCookiesButton = styled("button")`
  margin-right: 20px;
  font-size: ${fontSize};
  color: #0b0c0c;
  background-color: transparent;
  border: none;
  padding: 0;
  text-decoration: underline;
  cursor: pointer;
  text-align: left;
  &:hover {
    color: #4a4a4a;
  }
  &:visited {
    color: #2f2f2f;
  }
`;

const AnchorTag = styled(asAnchor("a"))`
  padding: 5px;
  font-size: ${fontSize};
  @media only screen and (min-width: 641px) {
    font-size: ${fontSize};
  }
`;

const CookieBanner = () => (
  <Banner id="cookieBanner">
    <BannerBody>
      <BannerParagraph mb={2}>
        Register a food business uses a cookie to create the registration. We
        also use optional cookies to analyse how the service is performing. We
        do not share any information with advertisers or social media platforms.
        We do not store any information about you in the cookie.
      </BannerParagraph>
      <BannerRow>
        <BannerLinkContainer>
          <AnchorTag id="cookieInfo" href="https://www.gov.uk/help/cookies">
            Find out more about cookies
          </AnchorTag>
          <AnchorTag
            id="cookiePolicy"
            href="https://www.food.gov.uk/cookie-policy"
          >
            Read our cookie policy
          </AnchorTag>
        </BannerLinkContainer>

        <BannerActionContainer>
          <form action="/setcookie/acceptAllCookies/false" method="post">
            <RejectCookiesButton id="cookieReject" type="submit">
              I do not accept optional cookies
            </RejectCookiesButton>
          </form>
          <form action="/setcookie/acceptAllCookies/true" method="post">
            <CookieButton id="cookieAccept" type="submit">
              I accept cookies
            </CookieButton>
          </form>
        </BannerActionContainer>
      </BannerRow>
    </BannerBody>
  </Banner>
);

export default CookieBanner;
