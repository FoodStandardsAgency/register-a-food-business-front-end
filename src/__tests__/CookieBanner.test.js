import CookieBanner from "../components/CookieBanner";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import * as emotion from "emotion";
import { createSerializer } from "jest-emotion";

expect.addSnapshotSerializer(createSerializer(emotion));

describe("<CookieBanner />", () => {
  it("renders without crashing", () => {
    const wrapper = shallow(<CookieBanner />);
    expect(wrapper.length).toBe(1);
  });

  it("matches the previous snapshot", () => {
    const tree = renderer.create(<CookieBanner />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
