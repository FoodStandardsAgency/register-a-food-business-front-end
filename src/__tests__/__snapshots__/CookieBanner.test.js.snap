// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CookieBanner /> matches the previous snapshot 1`] = `
.emotion-18 {
  background-color: #d9e8f2;
}

.emotion-16 {
  width: calc(100% - 30px);
  max-width: 960px;
  margin: auto;
  padding: 15px 0;
}

@media (min-width:641px) {
  .emotion-16 {
    width: calc(100% - 60px);
  }
}

.emotion-1 {
  font-family: "nta",Arial,sans-serif;
  margin: 0;
  font-size: 16px;
  line-height: 1.25;
  margin-bottom: 10px;
  font-size: 16px;
}

.emotion-1 > p {
  margin: 0;
}

.emotion-1 > p > code {
  padding: 0.2em 0.4em;
  margin: 0px;
  font-size: 85%;
  background-color: rgba(27,31,35,0.05);
  border-radius: 3px;
}

.emotion-1 > pre {
  padding: 16px;
  overflow: auto;
  font-size: 85%;
  line-height: 1.45;
  background-color: #f6f8fa;
  border-radius: 3px;
}

.emotion-1 > pre > code {
  display: inline;
  padding: 0px;
  margin: 0px;
  border: 0px;
  overflow: visible;
  line-height: inherit;
  word-wrap: normal;
}

@media only screen and (min-width:641px) {
  .emotion-1 {
    font-size: 19px;
    line-height: 1.3;
  }
}

@media only screen and (min-width:641px) {
  .emotion-1 {
    margin-bottom: 10px;
  }
}

@media only screen and (min-width:641px) {
  .emotion-1 {
    font-size: 16px;
  }
}

.emotion-14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

@media only screen and (min-width:641px) {
  .emotion-14 {
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-align-items: flex-end;
    -webkit-box-align: flex-end;
    -ms-flex-align: flex-end;
    align-items: flex-end;
  }
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin-right: 20px;
}

.emotion-2 {
  color: #005ea5;
  padding: 3px;
  margin: -3px;
  outline-color: transparent;
  display: inline-block;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  padding: 5px;
  font-size: 16px;
}

.emotion-2:focus {
  background-color: #ffbf47;
  outline: 3px solid #ffbf47;
}

.emotion-2:visited {
  color: #2e358b;
}

@media only screen and (min-width:641px) {
  .emotion-2 {
    font-size: 16px;
  }
}

.emotion-12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  margin-top: 10px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
}

@media only screen and (min-width:641px) {
  .emotion-12 {
    -webkit-flex-wrap: nowrap;
    -ms-flex-wrap: nowrap;
    flex-wrap: nowrap;
    -webkit-box-pack: end;
    -webkit-justify-content: flex-end;
    -ms-flex-pack: end;
    justify-content: flex-end;
    margin-top: 0px;
    -webkit-align-items: flex-end;
    -webkit-box-align: flex-end;
    -ms-flex-align: flex-end;
    align-items: flex-end;
  }
}

.emotion-8 {
  margin-right: 20px;
  font-size: 16px;
  color: #0b0c0c;
  background-color: transparent;
  border: none;
  padding: 0;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  cursor: pointer;
  text-align: left;
}

.emotion-8:hover {
  color: #4a4a4a;
}

.emotion-8:visited {
  color: #2f2f2f;
}

.emotion-11 {
  background-color: #00823b;
  border: none;
  box-shadow: 0 2px 0 #003618;
  color: #ffffff;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  font-family: "nta",Arial,sans-serif;
  font-weight: 400;
  font-size: 1em;
  line-height: 1.25;
  outline-offset: -1px;
  outline: 1px solid transparent;
  padding: .526315em .789473em .263157em;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-appearance: none;
  -webkit-font-smoothing: antialiased;
  opacity: 1;
  pointer-events: auto;
  padding-right: .84211em;
  margin-bottom: 15px;
  margin-bottom: 0px;
  background-color: white;
  box-shadow: 0 2px 0 #4e80b4;
  color: #0b0c0c;
  font-size: 16px;
}

.emotion-11:hover {
  background-color: #00692f;
  color: #ffffff;
}

.emotion-11:focus {
  color: #ffffff;
  background-color: #00692f;
  outline: 3px solid #ffbf47;
}

.emotion-11:active {
  position: relative;
  top: 2px;
  box-shadow: 0 0 0 #003618;
}

.emotion-11:visited {
  color: #00823b;
}

.emotion-11 svg {
  max-width: 15px;
  margin-left: 20px;
}

@media only screen and (min-width:641px) {
  .emotion-11 {
    margin-bottom: 20px;
  }
}

.emotion-11:hover {
  background-color: #f2f2f2;
  color: #0b0c0c;
}

.emotion-11:focus {
  background-color: #f2f2f2;
  color: #0b0c0c;
}

@media only screen and (min-width:641px) {
  .emotion-11 {
    font-size: 16px;
    margin-bottom: 0px;
  }
}

<div
  className="emotion-18 emotion-19"
  id="cookieBanner"
>
  <div
    className="emotion-16 emotion-17"
  >
    <div
      className="emotion-0 emotion-1"
    >
      <p>
        Register a food business uses a cookie to create the registration. We also use optional cookies to analyse how the service is performing. We do not share any information with advertisers or social media platforms. We do not store any information about you in the cookie.
      </p>
    </div>
    <div
      className="emotion-14 emotion-15"
    >
      <div
        className="emotion-6 emotion-7"
      >
        <a
          className="emotion-2 emotion-3"
          href="https://www.gov.uk/help/cookies"
          id="cookieInfo"
          target="_blank"
        >
          Find out more about cookies
        </a>
        <a
          className="emotion-2 emotion-3"
          href="https://www.food.gov.uk/cookie-policy"
          id="cookiePolicy"
          target="_blank"
        >
          Read our cookie policy
        </a>
      </div>
      <div
        className="emotion-12 emotion-13"
      >
        <form
          action="/setcookie/acceptAllCookies/false"
          method="get"
        >
          <button
            className="emotion-8 emotion-9"
            id="cookieReject"
            type="submit"
          >
            I do not accept optional cookies
          </button>
        </form>
        <form
          action="/setcookie/acceptAllCookies/true"
          method="get"
        >
          <button
            className="emotion-10 emotion-11"
            disabled={false}
            id="cookieAccept"
            type="submit"
          >
            I accept cookies
          </button>
        </form>
      </div>
    </div>
  </div>
</div>
`;
